src/index.ts

```diff
@@ -9,13 +33,100 @@ if (require('electron-squirrel-startup')) { // eslint-disable-line global-requir
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 require('update-electron-app')()
 
-const createWindow = () => {
+const loginSSO = async () => {
+  const awsCmd = new awsCli.Aws();
+
+  // For testing purposes instead of text entry boxes
+  const profile = "SandboxSSOAdmin";
+
+  // TODO add a text entry for this
+  const url = await awsCmd.command(`configure get sso_start_url --profile ${profile}`).then(function (resp){
+    return resp.raw.trim();
+  });
+
+  // TODO list roles and let user select
+  const roleName = await awsCmd.command(`configure get sso_role_name --profile ${profile}`).then(function (resp){
+    return resp.raw.trim();
+  });
+  // TODO add a text entry for this
+  const account = await awsCmd.command(`configure get sso_account_id --profile ${profile}`).then(function (resp){
+    return resp.raw.trim();
+  });
+  // TODO add a dropdown for region entry
+  const region = await awsCmd.command(`configure get sso_region --profile ${profile}`).then(function (resp){
+      return resp.raw.trim();
+  });
+
+  console.log(`${url} and ${region}`);
+  const ssoOidc = new aws.SSOOIDC({
+    region: region,
+  });
+
+  // Register the client
+  ssoOidc.registerClient({
+    clientName: 'murderHornet',
+    clientType: 'public'
+  }, function(regErr,regData) {
+    if (regErr) { console.log(regErr, regErr.stack);} // an error occurred
+    else {
+      console.log(regData);
+      ssoOidc.startDeviceAuthorization({
+        clientId:regData.clientId,
+        clientSecret:regData.clientSecret,
+        startUrl: url,
+      },
+      function(err,data) {
+        if (err) console.log(err, err.stack); // an error occurred
+        else   {
+          const ssoWindow = new BrowserWindow({
+            height: 400,
+            width: 400,
+          });
+          ssoWindow.loadURL(data.verificationUriComplete);
+          ssoWindow.on('close', function(){
+            console.log('SSO WINDOW CLOSED CHECK FOR TOKEN');
+            ssoOidc.createToken({
+              clientId:regData.clientId,
+              clientSecret:regData.clientSecret,
+              grantType: 'urn:ietf:params:oauth:grant-type:device_code',
+              deviceCode: data.deviceCode
+            }, function(tokenErr,tokenData) {
+              if (tokenErr) console.log(tokenErr, tokenErr.stack); // an error occurred
+              else {
+                console.log(tokenData);
+                cacheToken = tokenData;
+                const sso = new aws.SSO({region: region});
+                // TODO List Roles the user can select
```

Comment:
ooh maybe this can do both listAccounts and then listRoles on each account the user has access to? https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/SSO.html#listAccounts-property